blockchain/state_manager/src/lib.rs

```diff
@@ -437,6 +440,35 @@ where
         Ok((state_root, receipt_root))
     }
 
+    /// Handles state migrations. Returns the new state root if a migration was
+    /// run.
+    fn handle_state_migrations(
+        &self,
+        parent_state: &Cid,
+        epoch: ChainEpoch,
+    ) -> anyhow::Result<Option<Cid>> {
+        match epoch {
+            x if x == self.chain_config.epoch(Height::Hygge) => {
+                info!("Running Hygge migration at epoch {epoch}");
+                let start_time = time::Instant::now();
+                let new_state = forest_state_migration::run_nv18_migration(
+                    &self.chain_config,
+                    self.blockstore(),
+                    parent_state,
+                    epoch,
+                )?;
+                let elapsed = start_time.elapsed().as_secs_f32();
```

Comment:
Would it be helpful to add this to Prometheus as well?

Reply:
I am not sure if it's worth it. The migration happens once every two, even three months.